### ISSUE TITLE
🐛 [Bug]: Why do long requests block each other?

### ISSUE THREAD

**Issue by fanybook** [gofiber/fiber#3015]
### Bug Description

like  https://github.com/gin-gonic/gin/issues/3981

### How to Reproduce

```go
package main

import (
    "fmt"
    "time"

    "github.com/gofiber/fiber/v2"
)

func handler(c *fiber.Ctx) error {
    fmt.Println(998.1)
    time.Sleep(10 * time.Second)
    return c.SendString("Hello, Sleep!")
}

func handler2(c *fiber.Ctx) error {
    fmt.Println(998.2)
    return c.SendString("Hello, World!")
}

func main() {
    app := fiber.New()

    app.Get("/aaa", handler)
    app.Get("/bbb", handler2)

    // 启动 HTTP 服务器
    err := app.Listen(":3000")
    if err != nil {
        fmt.Println(err)
    }
}
```

### Expected Behavior

fix this bug

### Fiber Version

github.com/gofiber/fiber/v2 v2.26.0

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

**Comment by ReneWerner87** [comment]
```go
package main

import (
	"fmt"
	"log"
	"strconv"
	"time"

	"github.com/gofiber/fiber/v2"
)

func handler(c *fiber.Ctx) error {

	sleepTime, err := strconv.Atoi(c.Params("sleepTime"))
	if err != nil {
		return c.SendString("Error")
	}
	fmt.Println("handler", sleepTime)
	time.Sleep(time.Duration(sleepTime) * time.Second)
	return c.SendString("Hello, Sleep!")
}

func handler2(c *fiber.Ctx) error {
	fmt.Println("handler2")
	return c.SendString("Hello, World!")
}

func main() {
	app := fiber.New()

	app.Get("/aaa/:sleepTime", handler)
	app.Get("/bbb", handler2)

	log.Fatalln(app.Listen(":3000"))
}

```

the handler with the sleep is not blocked
try 
curl http://localhost:3000/aaa/10
and shortly after that
curl http://localhost:3000/aaa/1

-> the handler with the sleep 1 will response before the other handler

**Comment by ReneWerner87** [comment]
@fanybook so there is no blocking

**Comment by fanybook** [comment]
> @fanybook so there is no blocking

there is no blocking too after added query params.

eg. http://127.0.0.1:3000/aaa?some=1 and http://127.0.0.1:3000/aaa?some=2

but, if only the body is different (eg. post http://127.0.0.1:3000/aaa), it will still block

**Comment by ReneWerner87** [comment]
![image](https://github.com/gofiber/fiber/assets/7063188/bc2c0dea-ace7-4e6e-8385-f15fcc206c46)

```go
package main

import (
	"fmt"
	"log"
	"math/rand"
	"time"

	"github.com/gofiber/fiber/v2"
)

// curl http://localhost:3000/aaa
func handler(c *fiber.Ctx) error {
	// random sleep time
	sleepTime := rand.Intn(10-1) + 1 // 1-10
	fmt.Println("start time", sleepTime, time.Now().Format("2006-01-02 15:04:05"))
	time.Sleep(time.Duration(sleepTime) * time.Second)
	fmt.Println("finish with sleep time", sleepTime, time.Now().Format("2006-01-02 15:04:05"))

	return c.SendString("Hello, Sleep!")
}

func main() {
	app := fiber.New()

	app.Get("/aaa", handler)

	log.Fatalln(app.Listen(":3000"))
}
```
can you describe what you mean by blocking or send a screen video

you always use a 10s sleep in your handler which causes the completion to be delayed by 10s
which can lead to the impression that the next request is blocked, which is not the case

even with your example, if you have good output
with your 10s
![image](https://github.com/gofiber/fiber/assets/7063188/36b66ffa-4eb4-4cee-a480-b29821a81460)

**Comment by ReneWerner87** [comment]
@fanybook 
pls add `time.Now().Format("2006-01-02 15:04:05")` in your outputs

unfortunately, the description says nothing about when the curl is started and when it is finished

i have tested your code and it behaves as it should and no request blocks the other one
i can ping endpoint “aaa” and “bbb” several times in a row 

please provide a screen video or better output

**Comment by fanybook** [comment]
@ReneWerner87 

https://raw.githubusercontent.com/fanybook/static-repo/main/411_1716968205.mp4

**Comment by ReneWerner87** [comment]
unfortunately cannot reproduce it
can you try it on another system with the same code

something in your system is misconfigured or the golang setup is broken (try a new installation)

**Comment by ReneWerner87** [comment]
```go
package main

import (
    "fmt"
    "net/http"
    "time"
)

func handler(w http.ResponseWriter, r *http.Request) {
    fmt.Println(777.1, time.Now().Format("2006-01-02 15:04:05"))
    time.Sleep(10 * time.Second)
    w.Write([]byte("Hello, Sleep!"))
}

func handler2(w http.ResponseWriter, r *http.Request) {
    fmt.Println(777.2, time.Now().Format("2006-01-02 15:04:05"))
    w.Write([]byte("Hello, World!"))
}

func main() {
    http.HandleFunc("/aaa", handler)
    http.HandleFunc("/bbb", handler2)

    err := http.ListenAndServe(":3000", nil)
    if err != nil {
        fmt.Println(err)
    }
}

```
@fanybook try the net.http example of the standard golang library on your machine 
if it happens there too it should be due to your setup

**Comment by fanybook** [comment]
> ```go
> package main
> 
> import (
>     "fmt"
>     "net/http"
>     "time"
> )
> 
> func handler(w http.ResponseWriter, r *http.Request) {
>     fmt.Println(777.1, time.Now().Format("2006-01-02 15:04:05"))
>     time.Sleep(10 * time.Second)
>     w.Write([]byte("Hello, Sleep!"))
> }
> 
> func handler2(w http.ResponseWriter, r *http.Request) {
>     fmt.Println(777.2, time.Now().Format("2006-01-02 15:04:05"))
>     w.Write([]byte("Hello, World!"))
> }
> 
> func main() {
>     http.HandleFunc("/aaa", handler)
>     http.HandleFunc("/bbb", handler2)
> 
>     err := http.ListenAndServe(":3000", nil)
>     if err != nil {
>         fmt.Println(err)
>     }
> }
> ```
> 
> @fanybook try the net.http example of the standard golang library on your machine if it happens there too it should be due to your setup

@ReneWerner87 the net.http example of the standard golang library has same issue
[go-stdlib](https://raw.githubusercontent.com/fanybook/static-repo/main/4357d4113c6164dbb622dd8227bd0a7f.jpg)

like https://github.com/gin-gonic/gin/issues/3981

**Comment by ReneWerner87** [comment]
Then it's related to your system and set up and not to any framework

**Comment by ReneWerner87** [comment]
Can you also test with another http client

Postman, curl, wget, chrome

**Comment by fanybook** [comment]
> Can you also test with another http client
> 
> Postman, curl, wget, chrome

yes, you r right, Postman doesn't have this issue, it may just be a problem with Chrome

**Comment by ReneWerner87** [comment]
That means you only tested with Chrome and only with your own system and then directly opened 2 bug reports in different frameworks without testing various other things? 
Next time you should first rule out that it is not your system or test tools that are causing the problem. 
Otherwise you will make the users of the framework insecure/scared and ensure that time is blocked by the people of the respective framework. 
Please also close all other bugs
